django/db/migrations/serializer.py

```diff
@@ -0,0 +1,396 @@
+        value_repr = repr(self.value)
+        if isinstance(self.value, datetime_safe.date):
+            value_repr = "datetime.%s" % value_repr
+        return value_repr, {"import datetime"}
+
+
+class TimeSerializer(BaseSerializer):
+    def serialize(self):
+        value_repr = repr(self.value)
+        if isinstance(self.value, datetime_safe.time):
+            value_repr = "datetime.%s" % value_repr
+        return value_repr, {"import datetime"}
+
+
+class TimeDeltaSerializer(BaseSerializer):
+    def serialize(self):
+        return repr(self.value), {"import datetime"}
+
+
+class SettingsReferenceSerializer(BaseSerializer):
+    def serialize(self):
+        return "settings.%s" % self.value.setting_name, {"from django.conf import settings"}
+
+
+class IntegerTypeSerializer(BaseSerializer):
+    def serialize(self):
+        return repr(self.value), set()
+
+
+class BinaryTypeSerializer(BaseSerializer):
+    def serialize(self):
+        value_repr = repr(self.value)
+        if six.PY2:
+            # Prepend the `b` prefix since we're importing unicode_literals
+            value_repr = 'b' + value_repr
+        return value_repr, set()
+
+
+class TextTypeSerializer(BaseSerializer):
+    def serialize(self):
+        value_repr = repr(self.value)
+        if six.PY2:
+            # Strip the `u` prefix since we're importing unicode_literals
+            value_repr = value_repr[1:]
+        return value_repr, set()
+
+
+class DecimalSerializer(BaseSerializer):
+    def serialize(self):
+        return repr(self.value), {"from decimal import Decimal"}
+
+
+class DeconstructableSerializer(BaseSerializer):
+    @classmethod
+    def serialize_deconstructed(cls, path, args, kwargs):
+        name, imports = cls._serialize_path(path)
+        strings = []
+        for arg in args:
+            arg_string, arg_imports = serializer_factory(arg).serialize()
+            strings.append(arg_string)
+            imports.update(arg_imports)
+        for kw, arg in sorted(kwargs.items()):
+            arg_string, arg_imports = serializer_factory(arg).serialize()
+            imports.update(arg_imports)
+            strings.append("%s=%s" % (kw, arg_string))
+        return "%s(%s)" % (name, ", ".join(strings)), imports
+
+    @staticmethod
+    def _serialize_path(path):
+        module, name = path.rsplit(".", 1)
+        if module == "django.db.models":
+            imports = {"from django.db import models"}
+            name = "models.%s" % name
+        else:
+            imports = {"import %s" % module}
+            name = path
+        return name, imports
+
+    def serialize(self):
+        return self.serialize_deconstructed(*self.value.deconstruct())
+
+
+class ModelFieldSerializer(DeconstructableSerializer):
+    def serialize(self):
+        attr_name, path, args, kwargs = self.value.deconstruct()
+        return self.serialize_deconstructed(path, args, kwargs)
+
+
+class BaseManagerSerializer(ModelFieldSerializer):
```

Comment:
Can you please rename this one to `ModelManagerSerializer` and let it inherit from `DeconstructableSerializer`.